finality-aleph/src/environment.rs

```diff
@@ -131,18 +131,18 @@ impl<B: Block, H: Hash> UnitStore<B, H> {
 pub(crate) struct Environment<B: Block, H: Hash, C, BE, SC> {
     client: Arc<C>,
     select_chain: SC,
-    tx_consensus: Option<mpsc::UnboundedSender<NotificationIn<H>>>,
-    rx_consensus: Option<mpsc::UnboundedReceiver<NotificationOut<H>>>,
+    tx_consensus: mpsc::UnboundedSender<NotificationIn<H>>,
+    rx_consensus: mpsc::UnboundedReceiver<NotificationOut<H>>,
     tx_network: mpsc::UnboundedSender<NetworkCommand<B, H>>,
     rx_network: mpsc::UnboundedReceiver<NetworkEvent<B, H>>,
-    rx_order: Option<tokio::sync::mpsc::UnboundedReceiver<Vec<H>>>,
+    rx_order: Fuse<tokio::sync::mpsc::UnboundedReceiver<Vec<H>>>,
     auth_cryptostore: AuthorityKeystore,
     store: UnitStore<B, H>,
     requests: BinaryHeap<ScheduledTask<H>>,
-    request_ticker: time::Interval,
+    request_ticker: Fuse<time::Interval>,
```

Comment:
Changing the implementation of the Future trait for Environment to an `async` method `run_epoch(self)` requires usage of fused streams